File: ambeth/Ambeth.Util/ambeth/util/Lock.cs
using System;

namespace De.Osthus.Ambeth.Util
{
    public interface Lock
    {
        LockState ReleaseAllLocks();

        void ReacquireLocks(LockState lockState);

        void Lock();

        void Unlock();

        bool TryLock();

        bool TryLock(int millis);

        bool TryLock(TimeSpan timeSpan);

        bool IsReadLockHeld { get; }

        bool IsWriteLockHeld { get; }
    }
}

File: jambeth/jambeth-util/src/main/java/com/koch/ambeth/util/Lock.java
package com.koch.ambeth.util;

public interface Lock
{
	LockState releaseAllLocks();

	void reacquireLocks(LockState lockState);

	void lock();

	void unlock();

	boolean tryLock();

	boolean tryLock(long millis);

	boolean isReadLockHeld();

	boolean isWriteLockHeld();

	void lock(Thread lockOwningThread);

	void unlock(Thread lockOwningThread);
}
